lib/buffer.js

```diff
@@ -42,34 +42,92 @@ function alignPool() {
   }
 }
 
-
-function Buffer(arg, encoding) {
+/**
+ * The Buffer() construtor is "soft deprecated" ... that is, it is deprecated
+ * in the documentation and should not be used moving forward. Rather,
+ * developers should use one of the three new factory APIs: Buffer.from(),
+ * Buffer.allocUnsafe() or Buffer.alloc() based on their specific needs. There
+ * is no hard deprecation because of the extent to which the Buffer constructor
+ * is used in the ecosystem currently -- a hard deprecation would introduce too
+ * much breakage at this time. Eventually, however, we should re-evaluate that
+ * choice.
```

Comment:
Actually, it may have to happen before v6. Since using Uint8Array methods on Buffer may call the constructor due to inheritance changes in v8 to better match the ES6 spec, it'll expect it to be zero-filled.